### PR TITLE
:sparkles: Add additional sec group rule for additionalPorts of LB

### LINKED ISSUE #1588: When defining additionalPorts under apiServerLoadBalancer no security group rule is created (port opening)
/kind bug

**What steps did you take and what happened:**
When defining additionalPorts under apiServerLoadBalancer in the OpenStackCluster manifest no security group rule is created meaning the LB will not reach the requested port.

**What did you expect to happen:**
All ports added to additionalPorts to also have a corresponding security group rule added

**Anything else you would like to add:**

Sample manifest:
```yaml
apiVersion: v1
items:
- apiVersion: infrastructure.cluster.x-k8s.io/v1alpha6
  kind: OpenStackCluster
  metadata:
    labels:
      cluster.x-k8s.io/cluster-name: hux-lab1
    name: hux-lab1
  spec:
    allowAllInClusterTraffic: true
    apiServerLoadBalancer:
      additionalPorts:
      - 31991
      enabled: true
    cloudName: elastx
    controlPlaneEndpoint:
      host: REDACTED
      port: 6443
    disableAPIServerFloatingIP: false
    dnsNameservers:
    - 8.8.8.8
    - 4.4.4.4
    externalNetworkId: 600b8501-78cb-4155-9c9f-23dfcba88828
    identityRef:
      kind: Secret
      name: hux-lab1-cloud-config
    managedSecurityGroups: true
    network: {}
    nodeCidr: 10.128.0.0/22
    subnet: {}
```

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): 0.7.3
- Cluster-API version: 0.4.3
- OpenStack version: Ussuri
- Minikube/KIND version: Kind 0.17.0
- Kubernetes version (use `kubectl version`): 1.26.4 (also tested 1.27.2)
- OS (e.g. from `/etc/os-release`): Ubuntu 20.04

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1588 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Review by lentzi90 — COMMENTED** [review]
/lgtm

I guess we will have to start thinking about how to handle IPv6 for these also at some point (ref #1587), but I see this is consistent with the existing code.

**Comment by lentzi90** [comment]
/retitle :sparkles: Add additional sec group rule for additionalPorts of LB

File: pkg/cloud/services/networking/securitygroups_rules.go [hunk 1]
```diff
@@ -291,6 +291,35 @@ func GetSGWorkerAllowAll(remoteGroupIDSelf, secControlPlaneGroupID string) []inf
...
+			Direction:     "ingress",
+			EtherType:     "IPv4",
+			Protocol:      "tcp",
+			RemoteGroupID: remoteGroupIDSelf,
+		},
+		{
+			Description: "Additional ports",
+			Direction:   "ingress",
+			EtherType:   "IPv4",
+			Protocol:    "udp",
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply pkg/cloud/services/networking/securitygroups_rules.go:310]: Good point! :+1:

[the remark this answers is not in the document]
Comment (jichenjc) [reply pkg/cloud/services/networking/securitygroups_rules.go:309]: I removed the remote sec group, so it should be fit for all 
then add UDP as well, can you help review as well?

[the remark this answers is not in the document]
Comment (lentzi90) [reply pkg/cloud/services/networking/securitygroups_rules.go:309]: Looks good to me. @huxcrux is this ok?

[the remark this answers is not in the document]
Comment (huxcrux) [reply pkg/cloud/services/networking/securitygroups_rules.go:309]: Looks good to me :)

**Review by huxcrux — APPROVED** [review]
LGTM

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**